loyalty_partner_applicability/models/loyalty_rule.py

```diff
@@ -1,22 +1,15 @@
 # Copyright 2023 Tecnativa - Pilar Vargas
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
-from odoo import api, fields, models
 
+from odoo import models
 
-class LoyaltyRule(models.Model):
-    _inherit = "loyalty.rule"
-    _description = "Loyalty Rule"
-
-    rule_partners_domain = fields.Char(
-        string="Based on Customers",
-        help="Loyalty program will work for selected customers only",
-        default="[]",
-    )
 
-    @api.model_create_multi
-    def create(self, vals_list):
-        res = super().create(vals_list)
-        for vals in vals_list:
-            if not vals.get("rule_partners_domain", False):
-                vals["rule_partners_domain"] = "[]"
-        return res
+class LoyaltyRule(models.Model):
+    _name = "loyalty.rule"
+    _inherit = ["loyalty.rule", "loyalty.partner.applicability.mixin"]
```

Comment:
If the purpose of refactoring is to apply the rule at the promotion level, I find it a little confusing that a mixin is created and that it is still possible to apply this at the rule level. Ultimately, what is the approach? If it is applied at the promotion level, it would have to be removed at the rule level, as it could define different clients at two different levels, but only one would have priority, that of the rule. This would cause errors when configuring the rule and unexpected results when applying the promotion.